pkg/network/flag/networkflag.go

```diff
@@ -0,0 +1,51 @@
+package flag
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"github.com/weaveworks/ignite/pkg/network"
+)
+
+var plugins = network.ListPlugins()
+
+type NetworkPluginFlag struct {
+	value *network.PluginName
+}
+
+func (nf *NetworkPluginFlag) Set(val string) error {
+	var foundPlugin *network.PluginName
+	for _, plugin := range plugins {
+		if plugin.String() == val {
+			foundPlugin = &plugin
+			break
+		}
+	}
+	if foundPlugin == nil {
+		return fmt.Errorf("Invalid network mode %q, must be one of %v", val, plugins)
+	}
+	*nf.value = *foundPlugin
+	return nil
```

Comment:
If there's a return in the loop, you can just `return fmt.Errorf("invalid network mode %q, must be one of %v", val, plugins)` here